Contracts: Add invariant contract (#2800)
nitlang/nit: share/man/nitc.md

### `--full-contract`
Option used to enables contracts (ensures, expects) on all classes. Warning this is an expensive option at runtime.

### `--in-out-invariant`

> Enable the verification of invariant contracts in enter and exit (By default the invariant is only checked after the invocation).

Reviewer: ```suggestion
Enable the verification of invariant contracts on enter and exit (By default the invariant is only checked after the invocation).
```

> Enable the verification of invariant contracts on enter and exit (By default the invariant is only checked after the invocation).

Reviewer: ```suggestion
Enable invariants checking on enter and exit (by default invariants are only checked at exit).
```

### `--no-self-contract`

> Disable the verification of contracts in self.

Reviewer: Should have a bit more documentation as without context `self` doesn't mean much.

> Disable the verification of contracts when the receiver of the invocation is `self`.

Reviewer: ```suggestion
Disable contracts checking for methods called on `self`.
```
